File: src/main/locator.ts
import { existsSync } from 'node:fs'
import { homedir } from 'node:os'
import { join } from 'node:path'

const SAVE_FILENAME = 'SaveFile_Live.es3'

/**
 * Caminhos candidatos do save (ver TBHPEDIA.md > Save).
 * Windows: %USERPROFILE%\AppData\LocalLow\TesseractStudio\TaskbarHero\
 * Proton/Linux: compatdata/3678970/pfx/... (suporte futuro)
 */
export function candidateSavePaths(): string[] {
  const home = homedir()
  const paths: string[] = []

  // Windows (LocalLow nao tem env var dedicada; derivamos de USERPROFILE)
  paths.push(
    join(home, 'AppData', 'LocalLow', 'TesseractStudio', 'TaskbarHero', SAVE_FILENAME)
  )

  // Proton (Steam) no Linux
  const steamApp = '3678970'
  paths.push(
    join(
      home,
      '.steam',
      'steam',
      'steamapps',
      'compatdata',
      steamApp,
      'pfx',
      'drive_c',
      'users',
      'steamuser',
      'AppData',
      'LocalLow',
      'TesseractStudio',
      'TaskbarHero',
      SAVE_FILENAME
    )
  )

  return paths
}

/** Retorna o primeiro caminho de save existente, ou null. */
export function locateSave(): string | null {
  for (const p of candidateSavePaths()) {
    if (existsSync(p)) return p
  }
  return null
}

File: dotnet/TbhTracker.App/Services/Locator.cs
namespace TbhTracker.App.Services;

/// <summary>Port de src/main/locator.ts. Localiza o save do TBH.
/// Windows: %USERPROFILE%\AppData\LocalLow\TesseractStudio\TaskbarHero\SaveFile_Live.es3.</summary>
public sealed class Locator
{
    private const string SaveFilename = "SaveFile_Live.es3";

    public List<string> CandidateSavePaths()
    {
        var home = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
        var paths = new List<string>
        {
            Path.Combine(home, "AppData", "LocalLow", "TesseractStudio", "TaskbarHero", SaveFilename)
        };

        // Proton (Steam) no Linux — suporte futuro.
        const string steamApp = "3678970";
        paths.Add(Path.Combine(home, ".steam", "steam", "steamapps", "compatdata", steamApp,
            "pfx", "drive_c", "users", "steamuser", "AppData", "LocalLow",
            "TesseractStudio", "TaskbarHero", SaveFilename));

        return paths;
    }

    public string? LocateSave()
    {
        foreach (var p in CandidateSavePaths())
            if (File.Exists(p)) return p;
        return null;
    }
}
